test/Microsoft.AspNetCore.Http.Connections.Tests/HttpConnectionManagerTests.cs

```diff
@@ -28,6 +28,45 @@ public void NewConnectionsHaveConnectionId()
             Assert.NotNull(connection.Application);
         }
 
+        [Theory]
+        [InlineData(false, false, false)]
+        [InlineData(false, false, true)]
+        [InlineData(false, true, true)]
+        [InlineData(false, true, false)]
+        [InlineData(true, false, false)]
+        [InlineData(true, false, true)]
+        [InlineData(true, true, true)]
+        [InlineData(true, true, false)]
```

Comment:
![image](https://user-images.githubusercontent.com/303201/38417886-2fcee0c8-39ee-11e8-92b9-9bc26e1d3f9a.png)

Reply:
Maybe it's worth using separate `Fact`s with a helper method for these... the test names are going to be weird: `DisposingConnectionsClosesBothSidesOfThePipe(true, false, false)`